src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -125,5 +125,42 @@ public static ReadOnlySpan<char> Remove(this ReadOnlySpan<char> source, int star
             source.Slice(startIndex + count).CopyTo(result.Slice(startIndex));
             return result;
         }
+
+        public static int LastIndexOf(this ReadOnlySpan<char> source, char value)
+        {
+            if (source.Length == 0)
+                return -1;
+
+            for (int i = source.Length; i >= 0; i--)
+            {
+                if (source[i] == value)
+                    return i;
+            }
+            return -1;
+        }
+
+        public static int LastIndexOf(this ReadOnlySpan<char> source, char value, int startIndex)
```

Comment:
You should delete these two overloads that take startIndex (and count), and call Slice where these methods are called. It is the preferred way to write Span-based code.